### ISSUE TITLE
Implement HasStyle and HasSize in Component

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#9196]
### Description of the bug
```
A Component is a higher level abstraction of an {@link Element} or a hierarchy of {@link Element}s.
```

Each and every `Element` in a browser can:
* Add and remove class names (`HasStyle`)
* Set the size(`HasSize`)

`Component` should thus implement these two interfaces.

### Minimal reproducible example
What you cannot do today is e.g. set a class name to a collection of fields without casting:
```
AbstractField[] fields = new AbstractField[] { firstName, lastName, email, phone, dateOfBirth, occupation };
for (AbstractField field : fields) {
    ((HasStyle) field).addClassName("full-width");
}
```

### Expected behavior
Expected would be
```
Component[] fields = new Component[] { firstName, lastName, email, phone, dateOfBirth, occupation };
for (Component field : fields) {
    field.addClassName("full-width");
}
```

### Versions:
    - Vaadin version: 14

**Comment by Legioth** [comment]
That change would effectively make those interfaces redundant, so if this is implemented, then it should be considered to include the implementation in `Component` and deprecate those interfaces.

The point is not to express that it's possible to set the size of a component, but rather to express that the author of that component class sees that users of the component are expected to set the size of the component.

It's actually also not true that the size and class names can be set for every `Element`, since text nodes are also implemented to implement `Element` to avoid making everything else more complicated.

**Comment by Artur-** [comment]
> It's actually also not true that the size and class names can be set for every Element, since text nodes are also implemented to implement Element to avoid making everything else more complicated.

This is a moot argument unless you remove `setAttribute` and other methods from `Element`

>The point is not to express that it's possible to set the size of a component, but rather to express that the author of that component class sees that users of the component are expected to set the size of the component.

Are you optimizing for edge cases here? Which component would be such that a user should not set a class name on it?

**Comment by Legioth** [comment]
I don't think the comparison to `Element` makes sense for two reasons:
1. It's a low level API where it is more acceptable to have API that doesn't work in all cases
2. It's not designed to be subclassed, which means that everything must be in the main API.

> Which component would be such that a user should not set a class name on it?

`HasStyle` is indeed more applicable for any component since it's more about how the rest of the page relates to the component.

That's why I used `HasSize` as the example - it's not unreasonable to assume that there are components that are designed to always have specific dimensions. It's in my mind similar to how all components (except those backed by text nodes) can technically have child elements, but we're still not making `add(Component)` available in the `Component` API. `HasSize` is still more universal than adding child elements, but it's still questionable whether it's universal enough to make it available in `Component`.

**Comment by pleku** [comment]
I agree on sizing being more component specific thing than setting class names which used often for styling the components with css selectors. It used to be considered even a best practice to "always name your components with class names" to make the styling easier later on. For the sizing methods, exposing all (?) of those in `Component` could mean that in a case where those should not be used for a specific component, it would mean that a developer should have to deprecate and throw for all of them, which, while being an edge case, would still be PITA.
